src/EloquentBuilderTrait.php

```diff
@@ -13,83 +14,99 @@
             }
 
-            if (isset($sort)) {
-                if (!is_array($sort)) {
-                    throw new InvalidArgumentException('Sort should be an array.');
-                }
+            $queryBuilder->with($includes);
+        }
 
-                if (!isset($filterJoins)) {
-                    $filterJoins = [];
-                }
+        if (isset($filter_groups)) {
+            $filterJoins = $this->applyFilterGroups($queryBuilder, $filter_groups);
+        }
 
-                $sortingJoins = $this->applySorting($query, $sort, $filterJoins);
+        if (isset($sort)) {
+            if (!is_array($sort)) {
+                throw new InvalidArgumentException('Sort should be an array.');
             }
 
-            if (isset($limit)) {
-                $query->limit($limit);
+            if (!isset($filterJoins)) {
+                $filterJoins = [];
             }
 
-            if (isset($page)) {
-                $query->offset($page*$limit);
-            }
+            $sortingJoins = $this->applySorting($queryBuilder, $sort, $filterJoins);
+        }
+
+        if (isset($limit)) {
+            $queryBuilder->limit($limit);
+        }
+
+        if (isset($page)) {
+            $queryBuilder->offset($page*$limit);
         }
 
-        return $query;
+        return $queryBuilder;
     }
 
-    protected function applyFilterGroups(Builder $query, array $filterGroups = [], array $previouslyJoined = [])
+    /**
+     * @param Builder $queryBuilder
+     * @param array $filterGroups
+     * @param array $previouslyJoined
+     * @return array
+     */
+    protected function applyFilterGroups(Builder $queryBuilder, array $filterGroups = [], array $previouslyJoined = [])
     {
         $joins = [];
         foreach ($filterGroups as $group) {
             $or = $group['or'];
             $filters = $group['filters'];
 
-            $query->where(function (Builder $query) use ($filters, $or, &$joins) {
+            $queryBuilder->where(function (Builder $query) use ($filters, $or, &$joins) {
                 foreach ($filters as $filter) {
                     $this->applyFilter($query, $filter, $or, $joins);
                 }
             });
         }
 
         foreach(array_diff($joins, $previouslyJoined) as $join) {
-            $this->joinRelatedModelIfExists($query, $join);
+            $this->joinRelatedModelIfExists($queryBuilder, $join);
         }
 
         return $joins;
     }
 
-    protected function applyFilter(Builder $query, array $filter, $or = false, array &$joins)
+    /**
+     * @param Builder $queryBuilder
+     * @param array $filter
+     * @param bool|false $or
+     * @param array $joins
+     */
+    protected function applyFilter(Builder $queryBuilder, array $filter, $or = false, array &$joins)
     {
         // $value, $not, $key, $operator
         extract($filter);
 
-        $table = $query->getModel()->getTable();
+        $dbType = Config::get('database.default');
```

Comment:
I do not think this is the most optimal way to get the connection type. The database key can be any arbitrary value.

What about getting it from the queryBuilder, something like `$queryBuilder->getConnection()->getDriverName()`. I just quickly looked it up, so I am not 100% sure it is a viable option - could you look into it? :-)